### ISSUE TITLE
Skulpin rendering freezes when window geometry changes

### ISSUE THREAD

**Issue by Kethku** [aclysma/skulpin#47]
I've gotten reports in neovide and can repro myself that skulpin does not recover if various types of window geometry change.

The way I can reliably trigger this is by either opening neovide (or I think any skulpin app) in remote desktop on windows, disconnect from remote desktop and login to the machine locally. The previously responsive app will show only a black screen and not recover until restarted.

I've also seen this happen by connecting and disconnecting monitors. Further there are reports that this happens on linux as well as windows.

@aclysma do you have any ideas where to start looking to get this fixed? I'm down to do some investigation, and others have expressed interest in debugging, but I'm sorta at a loss given that I don't have a super good understanding of the renderer internals. Any clues would be greatly appreciated

**Comment by aclysma** [comment]
Do you have a log available? I'm assuming the swapchain needs to be recreated but maybe there is some sort of timing issue with that happening.

**Comment by Kethku** [comment]
Some extra info was gathered here: https://github.com/Kethku/neovide/issues/147. The crash stack trace is here: https://user-images.githubusercontent.com/1565930/81236128-5833e200-8fb1-11ea-8a56-c075ff2235ca.png

**Comment by Kethku** [comment]
I agree though. I think sculpin doesn't handle properly when the swapchain gets invalidated... or something. Like I said I'm not really an expert in this area

**Comment by Kethku** [comment]
```
[2020-05-06T23:23:47Z INFO  interactive_sdl2] Window { timestamp: 9936, window_id: 1, win_event: FocusLost }
[2020-05-06T23:23:47Z INFO  interactive_sdl2] TextEditing { timestamp: 9936, window_id: 1, text: "", start: 0, length: 0 }
[2020-05-06T23:23:54Z INFO  interactive_sdl2] Window { timestamp: 17423, window_id: 1, win_event: Leave }
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: "GetDpiForMonitor failed"', skulpin-renderer-sdl2\src\lib.rs:47:27
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
[2020-05-06T23:23:59Z DEBUG skulpin_renderer::renderer] destroying Renderer
[2020-05-06T23:23:59Z DEBUG skulpin_renderer::skia_renderpass] destroying VkSkiaRenderPass
[2020-05-06T23:23:59Z DEBUG skulpin_renderer::skia_renderpass] destroyed VkSkiaRenderPass
[2020-05-06T23:23:59Z DEBUG skulpin_renderer::swapchain] destroying VkSwapchain
[2020-05-06T23:23:59Z DEBUG skulpin_renderer::swapchain] destroyed VkSwapchain
[2020-05-06T23:23:59Z DEBUG skulpin_renderer::device] destroying VkDevice
[2020-05-06T23:23:59Z INFO  skulpin_renderer::debug_reporter] "Unloading layer library C:\\WINDOWS\\System32\\DriverStore\\FileRepository\\nvmd.inf_amd64_42bf84161ae379d8\\.\\nvoglv64.dll"
[2020-05-06T23:23:59Z INFO  skulpin_renderer::debug_reporter] "Unloading layer library C:\\VulkanSDK\\1.2.131.2\\Bin\\.\\VkLayer_khronos_validation.dll"
[2020-05-06T23:23:59Z DEBUG skulpin_renderer::device] destroyed VkDevice
[2020-05-06T23:23:59Z DEBUG skulpin_renderer::instance] destroying VkInstance
[2020-05-06T23:23:59Z DEBUG skulpin_renderer::debug_reporter] destroying VkDebugReporter
[2020-05-06T23:23:59Z DEBUG skulpin_renderer::debug_reporter] destroyed VkDebugReporter
[2020-05-06T23:23:59Z DEBUG skulpin_renderer::instance] destroyed VkInstance
[2020-05-06T23:23:59Z DEBUG skulpin_renderer::renderer] destroyed Renderer
error: process didn't exit successfully: `target\debug\examples\interactive_sdl2.exe` (exit code: 101)
```

Ooops. This looks like a bug in my code. Yikes

**Comment by Kethku** [comment]
Fixed in #49 Sorry for the confusion.

**Comment by aclysma** [comment]
Published fix in 0.9.4
